include/llama-cpp.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#ifndef __cplusplus
+#error "This header is for C++ only"
+#endif
+
+#include <memory>
+
+#include "llama.h"
+
+struct llama_model_deleter {
+    void operator()(llama_model * model) { llama_free_model(model); }
+};
+struct llama_context_deleter {
+    void operator()(llama_context * context) { llama_free(context); }
+};
+struct llama_sampler_deleter {
+    void operator()(llama_sampler * sampler) { llama_sampler_free(sampler); }
+};
+
+typedef std::unique_ptr<llama_model, llama_model_deleter> llama_model_ptr;
+typedef std::unique_ptr<llama_context, llama_context_deleter> llama_context_ptr;
+typedef std::unique_ptr<llama_sampler, llama_sampler_deleter> llama_sampler_ptr;
+typedef std::unique_ptr<char[]> char_array_ptr;
```

Comment:
The rest look good, but this one does not belong here.

Reply:
I also need to use more parameters as references, I notice I still pass raw pointers in places where it could be a reference to a smart pointer

Reply:
I removed char_array_ptr